packages/venia-ui/lib/RootComponents/Category/categoryContent.js

```diff
@@ -1,28 +1,37 @@
-import React, { Fragment, useCallback } from 'react';
+import React, { Fragment, Suspense, useCallback, useState } from 'react';
 import { shape, string } from 'prop-types';
 
-import { Title } from '../../components/Head';
+import { useAppContext } from '@magento/peregrine/lib/context/app';
+
 import { mergeClasses } from '../../classify';
-import FilterModal from '../../components/FilterModal';
+import { Title } from '../../components/Head';
+import Breadcrumbs from '../../components/Breadcrumbs';
 import Gallery from '../../components/Gallery';
 import Pagination from '../../components/Pagination';
 import defaultClasses from './category.css';
-import { useAppContext } from '@magento/peregrine/lib/context/app';
-import Breadcrumbs from '../../components/Breadcrumbs';
+
+const FilterModal = React.lazy(() => import('../../components/FilterModal'));
 
 // TODO: This can be replaced by the value from `storeConfig when the PR,
 // https://github.com/magento/graphql-ce/pull/650, is released.
 const pageSize = 6;
 const placeholderItems = Array.from({ length: pageSize }).fill(null);
 
 const CategoryContent = props => {
+    const { data, pageControl } = props;
+
     const [, { toggleDrawer }] = useAppContext();
+    const [loadFilters, setLoadFilters] = useState(false);
 
     const handleOpenFilters = useCallback(() => {
+        setLoadFilters(true);
         toggleDrawer('filter');
-    }, [toggleDrawer]);
+    }, [setLoadFilters, toggleDrawer]);
+
+    const handleLoadFilters = useCallback(() => {
+        setLoadFilters(true);
+    }, [setLoadFilters]);
```

Comment:
At some point we should add a talon for this component... I wonder why we didn't in the initial work :o

If you want to do that in this PR, that'd be awesome, but I wouldn't hold up this PR on it.

Reply:
I can do it here, np 👍

Reply:
Resolved in https://github.com/magento/pwa-studio/pull/1971/commits/a3ffd929180686bb97cc14f4e3f67a131ebc83fe.